### PR TITLE
Add tests to compiler.go

### PR DESCRIPTION
**Pull request by hachi8833**
To improve coverage.

### CODE REVIEW

File: compiler/compiler_test.go [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+		t.Fatalf("Expect `%d` for first instruction line. got: %d", e, a)
+	}
+
+	if e, a := 3, ci[0].Instructions[0].SourceLine(); e != a {
+		t.Fatalf("Expect `%d` for first instruction source line. got: %d", e, a)
+	}
+}
+
+func TestCompileToInstructionsNormalModePanic(t *testing.T) {
+	assert.Panics(t, func() {
```
[the remark this answers is not in the document]
Comment (hachi8833) [reply compiler/compiler_test.go]: I'll remove this

File: compiler/compiler_test.go [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+  99 + a
+end
+while true do
+end`, parser.NormalMode)
+
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+
+	if e, a := uint8(10), ci[0].Instructions[0].Opcode; e != a {
```
[the remark this answers is not in the document]
Comment (hachi8833) [reply compiler/compiler_test.go]: I'll rewrite thie

File: Gopkg.lock [hunk 3]
```diff
@@ -2,90 +2,155 @@
 
 
 [[projects]]
+  digest = "1:4337ac9fde4c6cb6b820c61e87d33a9e5e290398b14547ea1d544231bdcc0608"
```
[the remark this answers is not in the document]
Comment (hachi8833) [reply Gopkg.lock]: Sorry, this change was unintentional. I will revert this.

File: vendor/github.com/pmezard/go-difflib/LICENSE [hunk 4]
```diff
@@ -0,0 +1,27 @@
+Copyright (c) 2013, Patrick Mezard
```
Comment (st0012) [inline vendor/github.com/pmezard/go-difflib/LICENSE]: @hachi8833 can we remove this package?

File: compiler/bytecode/instruction.go [hunk 5]
```diff
@@ -116,11 +115,13 @@ func (i *Instruction) AnchorLine() int {
 }
 
 // Line returns instruction's line number
+// TODO: needs to change the func to simple public variable
```
Comment (st0012) [inline compiler/bytecode/instruction.go]: I don't think we need to add these TODO comments

File: vendor/github.com/pmezard/go-difflib/LICENSE (hunk 4 above)
Comment (hachi8833) [reply vendor/github.com/pmezard/go-difflib/LICENSE]: Removed

File: compiler/bytecode/instruction.go (hunk 5 above)
Comment (hachi8833) [reply compiler/bytecode/instruction.go]: Removed

**Pull request merged**